extractors-opendronemap/terraref-opendronemap.py

```diff
@@ -54,6 +63,68 @@ def check_delete_folder(folder):
             logging.debug("Execption deleting folder %s", folder)
             logging.debug("  %s", ex.message)
 
+def exif_tags_to_timestamp(exif_tags):
+    """Looks up the origin timestamp and a timestamp offset in the exit tags and returns
+       a datetime objext
+
+    Args:
+        exif_tags(dict): The exif tags to search for timestamp information
+
+    Return:
+        Returns the origin timestamp when found. The return timestamp is adjusted for UTF if
+        an offset is found. None is returned if a valid timestamp isn't found.
+    """
+    cur_stamp, cur_offset = (None, None)
+
+    def convert_and_clean_tag(value):
+        """Internal helper function for handling EXIF tag values. Tests for an empty string after
+           stripping colons, '+', '-', and whitespace [the spec is unclear if a +/- is needed when
+           the timestamp offset is unknown (and spaces are used)].
+        Args:
+            value(bytes or str): The tag value
+        Return:
+            Returns the cleaned up, and converted from bytes, string. Or None if the value is empty
+            after stripping above characters and whitespace.
+        """
+        if not value:
+            return None
+
+        # Convert bytes to string
+        if isinstance(value, bytes) and sys.version_info >= (3, 0):
+            value = value.decode('UTF-8').strip()
+        else:
+            value = value.strip()
+
+        # Check for an empty string after stripping colons
+        if value:
+            if not value.replace(":", "").replace("+:", "").replace("-", "").strip():
+                value = None
+
+        return None if not value else value
+
+    # Process the EXIF data
+    if EXIF_ORIGIN_TIMESTAMP in exif_tags:
+        cur_stamp = convert_and_clean_tag(exif_tags[EXIF_ORIGIN_TIMESTAMP])
+    if not cur_stamp:
+        return None
+
+    if EXIF_ORIGIN_TIMESTAMP_OFFSET in exif_tags:
+        cur_offset = convert_and_clean_tag(exif_tags[EXIF_ORIGIN_TIMESTAMP_OFFSET])
+    if not cur_offset and EXIF_TIMESTAMP_OFFSET in exif_tags:
+        cur_offset = convert_and_clean_tag(exif_tags[EXIF_TIMESTAMP_OFFSET])
+
+    # Format the string to a timestamp and return the result
+    try:
+        if not cur_offset:
+            cur_ts = datetime.datetime.strptime(cur_stamp, "%Y:%m:%d %H:%M:%S")
+        else:
+            cur_offset = cur_offset.replace(":", "")
+            cur_ts = datetime.datetime.strptime(cur_stamp + cur_offset, "%Y:%m:%d %H:%M:%S%z")
+    except Exception as ex:     # pylint: disable=broad-except
```

Comment:
As I mentioned elsewhere, this might be worth narrowing.

Reply:
The method already returns None so the caller should be able to handle that value. I'm leaving it up to the caller to determine what to do if a datetime is not available (for any reason).